File: 2020 lipiec/c++/1.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

string ident[200];

int main()
{
    ifstream in("identyfikator.txt");
    ofstream out("wyniki4_1.txt");
    
    for (int i = 0; i < 200; i++)
        in >> ident[i];
    
    int ans = 0;
    for (int i = 0; i < 200; i++)
    {
        int cur = 0;
        for (int j = 3; j < ident[i].size(); j++)
            cur += ident[i][j] - '0';

        if (cur > ans)
            ans = cur;
    }

    for (int i = 0; i < 200; i++)
    {
        int cur = 0;
        for (int j = 3; j < ident[i].size(); j++)
            cur += ident[i][j] - '0';

        if (cur == ans)
            out << ident[i] << '\n';
    }
}

File: 2020 lipiec/python/1.py
with open("identyfikator.txt", "r") as f:
    ident = f.read().strip().split('\n')

best = 0
for id in ident:
    cur = 0
    for i in id[3:]:
        cur += int(i)
    if cur > best:
        best = cur

with open("wyniki4_1.txt", "w") as f:
    for id in ident:
        cur = 0
        for i in id[3:]:
            cur += int(i)
        if cur == best:
            f.write(id + '\n')